layers/shader_object/shader_object.cpp

```diff
@@ -3497,6 +3507,22 @@ static VKAPI_ATTR PFN_vkVoidFunction VKAPI_CALL GetDeviceProcAddr(VkDevice devic
     return vtable.GetDeviceProcAddr(device, pName);
 }
 
+static VKAPI_ATTR PFN_vkVoidFunction VKAPI_CALL _layerGetPhysicalDeviceProcAddr(VkInstance instance, const char* pName) {
+    // See if this is a proc we want to intercept
+    if (void* func = FindFunctionByName(pName, kPhysicalDeviceFunctions)) {
+        return reinterpret_cast<PFN_vkVoidFunction>(func);
+    }
+    if (FindFunctionByName(pName, kInstanceFunctions | kDeviceFunctions)) {
+        return nullptr;
+    }
```

Comment:
I never noticed that the loader docs say layers should return NULL if it is *not* a physical device function. This is one of those "layers shouldn't be doing this...", but at the same time, it is good to have insurance that non-physical device functions are correctly identified. The loader checks all 'known' functions before calling GetPhysicalDeviceProcAddr, but unknown device/instance functions aren't known, so the loader can't do that :D